src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1328 @@
+        return $this->executeJsOnXpath($xpath, 'return arguments[0].innerHTML;');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOuterHtml(
+        #[Language('XPath')]
+        $xpath
+    ): string {
+        return $this->executeJsOnXpath($xpath, 'return arguments[0].outerHTML;');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAttribute(
+        #[Language('XPath')]
+        $xpath,
+        $name
+    ): ?string {
+        $escapedName = $this->jsonEncode($name, 'get attribute', 'attribute name');
+        $script = "return arguments[0].getAttribute($escapedName)";
+
+        return $this->executeJsOnXpath($xpath, $script);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getValue(
+        #[Language('XPath')]
+        $xpath
+    ) {
+        $element = $this->findElement($xpath);
+        $elementName = strtolower($element->getTagName() ?? '');
+        $elementType = strtolower((string)$element->getAttribute('type'));
+
+        // Getting the value of a checkbox returns its value if selected.
+        if ($elementName === 'input' && $elementType === 'checkbox') {
+            return $element->isSelected() ? $element->getAttribute('value') : null;
+        }
+
+        if ($elementName === 'input' && $elementType === 'radio') {
+            $script = <<<JS
+                var node = arguments[0],
+                    value = null;
+
+                var name = node.getAttribute('name');
+                if (name) {
+                    var fields = window.document.getElementsByName(name),
+                        i, l = fields.length;
+                    for (i = 0; i < l; i++) {
+                        var field = fields.item(i);
+                        if (field.form === node.form && field.checked) {
+                            value = field.value;
+                            break;
+                        }
+                    }
+                }
+
+                return value;
+                JS;
+
+            return $this->executeJsOnElement($element, $script);
+        }
+
+        // Using $element->attribute('value') on a select only returns the first selected option
+        // even when it is a multiple select, so a custom retrieval is needed.
+        if ($elementName === 'select' && $element->getAttribute('multiple')) {
+            $script = <<<JS
+                var node = arguments[0],
+                    value = [];
+
+                for (var i = 0; i < node.options.length; i++) {
+                    if (node.options[i].selected) {
+                        value.push(node.options[i].value);
+                    }
+                }
+
+                return value;
+                JS;
+
+            return $this->executeJsOnElement($element, $script);
+        }
+
+        // use textarea.value rather than textarea.getAttribute(value) for chrome 91+ support
+        if ($elementName === 'textarea') {
+            $script = <<<JS
```

Comment:
this should actually use `$element->getDomProperty('value')` on W3C-compliant Webdriver (probably also for inputs btw)